Add ViewGroup extension to allow concisely modifying of the layoutParams

steurt opened the pull request:
Added ViewGroup extension to allow concisely modifying of the layoutParams.

This allows following usage:
```
viewGroup.layoutParamsAs<ViewGroup.MarginLayoutParams> {
    updateMarginsRelative(start = marginStart)
}
```

Instead of:
```
val marginLayoutParams = viewGroup.layoutParams as ViewGroup.MarginLayoutParams
marginLayoutParams.updateMarginsRelative(start = 16)
viewGroup.layoutParams = marginLayoutParams
```

Was unable to update the `api` due to reified generics issue https://github.com/android/android-ktx/issues/235.

steurt commented:
I signed it!

steurt commented:
I signed it!

steurt commented:
I signed it!

steurt commented:
I signed it again!

romainguy commented on src/main/java/androidx/view/ViewGroup.kt:
```diff
@@ -77,6 +77,21 @@ val ViewGroup.children: Sequence<View>
+ * @see ViewGroup.getLayoutParams
+ * @see ViewGroup.setLayoutParams
+ **/
+inline fun <reified T> ViewGroup.layoutParamsAs(
```
The name does not match the behavior. I would expect this method to simply be a get+cast, not to trigger a requestLayout via setting the LPs.

steurt replied:
Do you think `withLayoutParams` would be better? Suggestions?

MGaetan89 replied:
`updateLayoutParams`?

steurt replied:
@romainguy would `updateLayoutParams` or `updateLayoutParamsAs` work for you?

steurt commented:
Renamed from `layoutParamsAs` to `updateLayoutParamsAs`. The usage then becomes:
```
viewGroup.updateLayoutParamsAs<ViewGroup.MarginLayoutParams> {
    updateMarginsRelative(start = marginStart)
}
```

Added non-generic variant `updateLayoutParams` for simply updating the layout params as `ViewGroup.LayoutParams`. This allows following usage:
```
viewGroup.updateLayoutParams { width = 500 }
```

steurt commented:
Renamed from `layoutParamsAs` to `updateLayoutParamsAs`. The usage then becomes:
```
viewGroup.updateLayoutParamsAs<ViewGroup.MarginLayoutParams> {
    updateMarginsRelative(start = marginStart)
}
```

Added non-generic variant `updateLayoutParams` for simply updating the layout params as `ViewGroup.LayoutParams`. This allows following usage:
```
viewGroup.updateLayoutParams { width = 500 }
```

steurt commented:
@romainguy any thoughts?

steurt commented:
I can name the generic version `updateLayoutParams` as well. However, in doing so the non-generic and generic variant will have the same JVM signature, i.e. a platform declaration clash.

This can be solved by specifying a different `@JvmName` annotation for each.
Perhaps:
```
@JvmName("updateViewGroupLayoutParams")
inline fun ViewGroup.updateLayoutParams(block: ViewGroup.LayoutParams.() -> Unit)

@JvmName("updateLayoutParams")
inline fun <reified T> ViewGroup.updateLayoutParams(block: T.() -> Unit)
```

Will this do?

romainguy commented:
Should we even have the non generic version at this point? @JakeWharton?

JakeWharton commented:
Playing with it locally.

JakeWharton commented:
How about this? https://github.com/android/android-ktx/compare/steurt-master

steurt commented:
@JakeWharton seems like a good setup to me!

romainguy commented:
LGTM
